src/Components/App.js

```diff
@@ -1,14 +1,55 @@
-import React, { Fragment } from 'react';
+/* eslint-disable react/no-unused-state */
+import React, { Component } from 'react';
 import Display from './Display';
 import ButtonPanel from './ButtonPanel';
+import Calculate from '../Logic/calculate';
 
-function App() {
-  return (
-    <>
-      <Display />
-      <ButtonPanel />
-    </>
-  );
-}
+export default class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: null,
+      next: null,
+      operation: null,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(buttonName) {
+    const { total, next, operation } = this.state;
+    const data = { total, next, operation };
+    const result = Calculate(data, buttonName);
+    this.setState({
+      total: result.total,
+      next: result.next,
+      operation: result.operation,
+    });
+  }
 
-export default App;
+  render() {
+    const { total, next, operation } = this.state;
+    let result;
+    if (operation === null) {
+      result = total;
+    } else if (operation === '+/-') {
+      if (next === null || next === '0') {
+        result = total;
+      } else {
+        result = next;
+      }
+    } else if (operation === '%') {
+      result = operation ? total : `${total}%`;
+    } else if (operation !== '%') {
+      result = next ? total + operation + next : total + operation;
```

Comment:
# [Optional]
### Heads-up:

There is `logic/calculate.js` file that you're supposed to have this logic, moreover, I think you already have it there, so you don't need to put it in here, and also these pieces of code are supposed to be implemented in the next milestone.

Reply:
Done. Understood